src/rust/src/x509/sct.rs

```diff
@@ -49,11 +49,72 @@ pub(crate) enum LogEntryType {
     PreCertificate,
 }
 
+#[derive(Clone)]
+pub(crate) enum HashAlgorithm {
+    None,
+    Md5,
+    Sha1,
+    Sha224,
+    Sha256,
+    Sha384,
+    Sha512,
+}
+
+impl TryFrom<u8> for HashAlgorithm {
+    type Error = PyAsn1Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        Ok(match value {
+            0 => HashAlgorithm::None,
+            1 => HashAlgorithm::Md5,
+            2 => HashAlgorithm::Sha1,
+            3 => HashAlgorithm::Sha224,
+            4 => HashAlgorithm::Sha256,
+            5 => HashAlgorithm::Sha384,
+            6 => HashAlgorithm::Sha512,
+            _ => {
+                return Err(PyAsn1Error::from(pyo3::exceptions::PyValueError::new_err(
+                    format!("Invalid HashAlgorithm variant ({})", value),
+                )))
+            }
+        })
+    }
+}
+
+#[derive(Clone)]
+pub(crate) enum SignatureAlgorithm {
+    Anonymous,
+    Rsa,
+    Dsa,
+    Ecdsa,
+}
+
+impl TryFrom<u8> for SignatureAlgorithm {
+    type Error = PyAsn1Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        Ok(match value {
+            0 => SignatureAlgorithm::Anonymous,
+            1 => SignatureAlgorithm::Rsa,
+            2 => SignatureAlgorithm::Dsa,
+            3 => SignatureAlgorithm::Ecdsa,
+            _ => {
+                return Err(PyAsn1Error::from(pyo3::exceptions::PyValueError::new_err(
+                    format!("Invalid SignatureAlgorithm variant ({})", value),
+                )))
+            }
+        })
+    }
+}
```

Comment:
Let me know if this is overkill. I can remove these crate-only enums and inline their bodies into their respective `Sct` APIs instead.